From a381c47d6fcad54d4499ba5cb5ccfc2e396696a4 Mon Sep 17 00:00:00 2001
From: Patrick Rudolph <patrick.rudolph@9elements.com>
Date: Wed, 30 Aug 2023 13:21:59 +0200
Subject: [PATCH 03/13] configurations: sbp1: Update PID settings

Change settings as requested by IBM.

- Update integral coefficient: -9.2 -> -5
- Update negative hysteresis: 5 -> 0
- Update proportional coefficient: -536 -> -500
- Update ILimitMax and OutLimitMax: 32000 -> 18000
- Update setpoints to higher limits for all groups
- Drop QSFP as the ADC temperature sensors should not be
  used to measure the QFSP temperature
- Set fans out limit minimum to 21% as 18% was found to be
  to low for some rare cases

Change-Id: I9b0d3cdc1c538b2d534225b3c5224e846a17cb30
Signed-off-by: Patrick Rudolph <patrick.rudolph@9elements.com>
---
 configurations/sbp1_chassis.json | 172 +++++++++++++------------------
 1 file changed, 72 insertions(+), 100 deletions(-)

diff --git a/configurations/sbp1_chassis.json b/configurations/sbp1_chassis.json
index 83fe374..4a09ace 100644
--- a/configurations/sbp1_chassis.json
+++ b/configurations/sbp1_chassis.json
@@ -925,7 +925,7 @@
             "Name": "Fan01_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan01_Pwm"
             ],
@@ -952,7 +952,7 @@
             "Name": "Fan02_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan02_Pwm"
             ],
@@ -979,7 +979,7 @@
             "Name": "Fan03_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan03_Pwm"
             ],
@@ -1006,7 +1006,7 @@
             "Name": "Fan04_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan04_Pwm"
             ],
@@ -1033,7 +1033,7 @@
             "Name": "Fan05_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan05_Pwm"
             ],
@@ -1060,7 +1060,7 @@
             "Name": "Fan06_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan06_Pwm"
             ],
@@ -1087,7 +1087,7 @@
             "Name": "Fan07_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan07_Pwm"
             ],
@@ -1114,7 +1114,7 @@
             "Name": "Fan08_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan08_Pwm"
             ],
@@ -1141,7 +1141,7 @@
             "Name": "Fan09_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan09_Pwm"
             ],
@@ -1168,7 +1168,7 @@
             "Name": "Fan10_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan10_Pwm"
             ],
@@ -1195,7 +1195,7 @@
             "Name": "Fan11_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan11_Pwm"
             ],
@@ -1222,7 +1222,7 @@
             "Name": "Fan12_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan12_Pwm"
             ],
@@ -1249,7 +1249,7 @@
             "Name": "Fan13_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan13_Pwm"
             ],
@@ -1276,7 +1276,7 @@
             "Name": "Fan14_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan14_Pwm"
             ],
@@ -1303,7 +1303,7 @@
             "Name": "Fan15_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan15_Pwm"
             ],
@@ -1330,7 +1330,7 @@
             "Name": "Fan16_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan16_Pwm"
             ],
@@ -1357,7 +1357,7 @@
             "Name": "Fan17_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan17_Pwm"
             ],
@@ -1384,7 +1384,7 @@
             "Name": "Fan18_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan18_Pwm"
             ],
@@ -1411,7 +1411,7 @@
             "Name": "Fan19_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan19_Pwm"
             ],
@@ -1438,7 +1438,7 @@
             "Name": "Fan20_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan20_Pwm"
             ],
@@ -1465,7 +1465,7 @@
             "Name": "Fan21_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan21_Pwm"
             ],
@@ -1492,7 +1492,7 @@
             "Name": "Fan22_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan22_Pwm"
             ],
@@ -1519,7 +1519,7 @@
             "Name": "Fan23_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan23_Pwm"
             ],
@@ -1546,7 +1546,7 @@
             "Name": "Fan24_PID",
             "NegativeHysteresis": 0,
             "OutLimitMax": 100,
-            "OutLimitMin": 18,
+            "OutLimitMin": 21,
             "Outputs": [
                 "Fan24_Pwm"
             ],
@@ -1564,8 +1564,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "BCM0_TEMP Temperature",
@@ -1574,41 +1574,13 @@
                 "BCM3_TEMP Temperature"
             ],
             "Name": "PSU Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
-            "OutLimitMin": 2500,
-            "Outputs": [],
-            "PCoefficient": -536,
-            "PositiveHysteresis": 0,
-            "SetPoint": 55,
-            "SlewNeg": 0,
-            "SlewPos": 0,
-            "Type": "Pid",
-            "Zones": [
-                "PSU"
-            ]
-        },
-        {
-            "Class": "temp",
-            "FFGainCoefficient": 0,
-            "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
-            "ILimitMin": 2500,
-            "Inputs": [
-                "QSFP1 Temperature",
-                "QSFP2 Temperature",
-                "QSFP3 Temperature",
-                "QSFP4 Temperature"
-            ],
-            "Name": "QSFP Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 95,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1620,8 +1592,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "Die_CPU1",
@@ -1630,13 +1602,13 @@
                 "Die_CPU4"
             ],
             "Name": "CPU Temperature",
-            "NegativeHysteresis": 2,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 65,
+            "SetPoint": 85,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1648,8 +1620,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "PVCCFA_EHV_CPU0 Temperature",
@@ -1674,13 +1646,13 @@
                 "PVPP_HBM_CPU3 Temperature"
             ],
             "Name": "CPU VR Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 85,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1692,8 +1664,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "P1V2_NIC Temperature",
@@ -1703,13 +1675,13 @@
                 "PVCORE_NIC2 Temperature"
             ],
             "Name": "NIC VR Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 85,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1721,8 +1693,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "DIMM_A1_CPU1",
@@ -1743,13 +1715,13 @@
                 "DIMM_H2_CPU1"
             ],
             "Name": "CPU1 VR Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 60,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1761,8 +1733,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "DIMM_A1_CPU2",
@@ -1783,13 +1755,13 @@
                 "DIMM_H2_CPU2"
             ],
             "Name": "CPU2 VR Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 60,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1801,8 +1773,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "DIMM_A1_CPU3",
@@ -1823,13 +1795,13 @@
                 "DIMM_H2_CPU3"
             ],
             "Name": "CPU3 VR Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 60,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1841,8 +1813,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "DIMM_A1_CPU4",
@@ -1863,13 +1835,13 @@
                 "DIMM_H2_CPU4"
             ],
             "Name": "CPU4 VR Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 60,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
-- 
2.41.0

